src/zarr/core/array.py

```diff
@@ -315,15 +313,9 @@ async def _create_v2(
             chunks=chunks,
             order=order,
             dimension_separator=dimension_separator,
-            fill_value=0 if fill_value is None else fill_value,
```

Comment:
Restoring the `0 if fill_value is None else fill_value` does fix the failure in tests/v3/test_array.py::test_serializable_sync_array.

But that said, I like the change here... Setting a default fill value *here* seems wrong.

Reply:
Oh, but this is just for `v2` metadata, which is allowed to be None. Maybe this isn't the wrong spot to do that.